File: examples/test_websocket.py
#!/usr/bin/env python3
"""
Simple WebSocket test client for Wolf server
Requires: pip install websockets
"""

import asyncio
import websockets

async def test_websocket():
    print("🐺 Wolf WebSocket Test Client\n")
    print("Connecting to ws://localhost:8080...\n")
    
    try:
        async with websockets.connect('ws://localhost:8080') as websocket:
            print("✅ Connected successfully!\n")
            
            # Send test message
            test_msg = "Hello, Wolf Server!"
            print(f"📤 Sending: {test_msg}")
            await websocket.send(test_msg)
            
            # Receive echo
            response = await websocket.recv()
            print(f"📨 Received: {response}\n")
            
            # Send more messages
            for i in range(1, 4):
                msg = f"Test message {i}"
                print(f"📤 Sending: {msg}")
                await websocket.send(msg)
                
                response = await websocket.recv()
                print(f"📨 Received: {response}\n")
                
                await asyncio.sleep(0.5)
            
            print("✅ All tests passed!")
            
    except Exception as e:
        print(f"❌ Error: {e}")
        return 1
    
    return 0

if __name__ == "__main__":
    exit(asyncio.run(test_websocket()))

File: examples/test_websocket.js
#!/usr/bin/env node

// Simple WebSocket test client for Wolf server
const WebSocket = require('ws');

console.log('🐺 Wolf WebSocket Test Client\n');
console.log('Connecting to ws://localhost:8080...\n');

const ws = new WebSocket('ws://localhost:8080');

ws.on('open', function open() {
    console.log('✅ Connected successfully!');
    console.log('Sending test message...\n');
    ws.send('Hello, Wolf Server!');
});

ws.on('message', function incoming(data) {
    console.log('📨 Received:', data.toString());
    
    // Send a few more test messages
    setTimeout(() => {
        console.log('\nSending: "Test message 2"');
        ws.send('Test message 2');
    }, 1000);
    
    setTimeout(() => {
        console.log('Sending: "Final test"');
        ws.send('Final test');
    }, 2000);
    
    setTimeout(() => {
        console.log('\n✅ All tests passed!');
        console.log('Closing connection...');
        ws.close();
    }, 3000);
});

ws.on('error', function error(err) {
    console.error('❌ Error:', err.message);
    process.exit(1);
});

ws.on('close', function close() {
    console.log('\n🔌 Connection closed');
    process.exit(0);
});

// Timeout after 10 seconds
setTimeout(() => {
    console.error('\n❌ Timeout - connection failed');
    process.exit(1);
}, 10000);
